src/domain/event/types.ts

```diff
@@ -1,4 +1,27 @@
+import {
+  CourseFieldsFragment,
+  EventFieldsFragment,
+} from '../../generated/graphql';
+import { ROUTES } from '../app/routes/constants';
+
 export type KeywordOption = {
   id: string;
   name: string;
 };
+
+export type EventFields = EventFieldsFragment | CourseFieldsFragment;
+
+export enum EventType {
```

Comment:
Using just type might be easier than enum?
```suggestion
export type EventType = 'event' | 'course';

export const EVENT_ROUTE_MAPPER: Record<EventType, string> = {
  event: ROUTES.EVENT,
  course: ROUTES.COURSE,
};

```

Reply:
it is expected to be enum, it is also used in other places:

<img width="693" alt="Screenshot 2020-12-31 at 15 01 14" src="https://user-images.githubusercontent.com/15219142/103411645-b97c7700-4b79-11eb-9c09-558f897cf35c.png">

Reply:
Just put `eventType="course"` there? If you put illegal string value there, ts will complain.

Reply:
hmm good point, that might be better solution